### PR TITLE
Remove hack for IntelliJ IDEs, replace with caveat

### PR DESCRIPTION
**Pull request by radeksimko**
### Related issues:
- #9572 
- #6910 
### TL;DR reasons:
- hack was causing issues during upgrades
- vendor officially doesn't support Java >6 anyway
### Details:

@w7tek in https://github.com/caskroom/homebrew-cask/issues/9572#issuecomment-77729696

> Unfortunately, according to the conversation along the same issue, it appears the idea.properties is picked up from a different location when using the CE than when using Ultimate.

True. I've been just playing around and I can confirm that this hack

```
echo "JVMVersion=1.6+" > ~/Library/Preferences/IntelliJIdea14/idea.properties
```

only works for _IntelliJ IDEA 14_ and _AppCode_, but doesn't for any other IDE (not even IntelliJ IDEA 14 CE). Maybe they have add this functionality into IntelliJ core and AppCode has already adopted it, so other IDEs may do in the future as well?

I wonder what's the motivation of JetBrains for being stuck 4 years behind (that's when Java 7 was released)... :confused:

### CODE REVIEW

**Comment by ctwise** [comment]
"I wonder what's the motivation of JetBrains for being stuck 4 years behind (that's when Java 7 was released)..."

It's due to many GUI defects still outstanding in Java 7, 8 and in previews of 9 as well. Oracle doesn't seem to care about AWT/SWT anymore and have moved on to JavaFX. As a result outstanding GUI issues in OS/X aren't getting addressed (and obviously haven't been for a very long time). JetBrains is apparently trying to create a patched JVM that deals with at least the worst problems and is planning on bundling it with the IDEs. You can obviously use Java 7+ to run the IDEs but JetBrains gets the same defect reports over and over again when running on those.

**Comment by msheakoski** [comment]
This patch should also be applied to [0xdbe-eap](https://github.com/caskroom/homebrew-cask/blob/master/Casks/0xdbe-eap.rb). When using it with JVM>6, the font rendering on OSX is horrible.

**Comment by sebroeder** [comment]
We should remove the plist hack from the JetBrains formulas in the `homebrew-versions` repository, too. caskroom/homebrew-versions#847 could help with this for the eap versions, if accepted by the maintainers.
